contracts/rust/deployer/src/lib.rs

```diff
@@ -762,6 +922,71 @@ pub async fn is_contract(provider: impl Provider, address: Address) -> Result<bo
     Ok(true)
 }
 
+pub async fn get_proxy_initialized_version(
+    provider: impl Provider,
+    proxy_addr: Address,
+) -> Result<u8> {
+    // From openzeppelin Initializable.sol, the initialized version slot is keccak256("openzeppelin.storage.Initializable");
+    let slot: B256 = "0xf0c57e16840df040f15088dc2f81fe391c3923bec73e23a9662efc9c229c6a00"
+        .parse()
+        .unwrap();
+    let value = provider.get_storage_at(proxy_addr, slot.into()).await?;
+    let initialized = value.as_le_bytes()[0]; // `_initialized` is u8 stored in the last byte
+    Ok(initialized)
+}
+
+/// this depends on upgradeProxy.ts which has to be ran on a real network supported by the safeSDK
+pub async fn call_upgrade_proxy_script(
+    proxy_addr: Address,
+    new_impl_addr: Address,
+    init_data: String,
+    rpc_url: String,
+    safe_addr: Address,
+    dry_run: Option<bool>,
+) -> Result<(String, bool), anyhow::Error> {
+    let dry_run = dry_run.unwrap_or(false);
+    tracing::info!("Dry run: {}", dry_run);
+    tracing::info!("Sending the upgrade proposal to multisig: {}", safe_addr);
+
+    let script_path = if let Ok(env_path) = std::env::var("MULTISIG_UPGRADE_ENTRYPOINT_PATH") {
```

Comment:
I'm not sure we need to this env var here. I thought we assured that the entrypoint script is in the path.

Reply:
thanks for this, it's not
https://github.com/EspressoSystems/espresso-network/commit/a5083dc31cde57a300c3c2d2296404a8953c75b5